test/Basic/matrix_single_subscript_load.test

```diff
@@ -0,0 +1,71 @@
+#--- source.hlsl
+RWBuffer<int> In : register(u0);
+RWBuffer<int> Out : register(u1);
+
+[numthreads(1,1,1)]
+void main() {
+    int4x4 A = int4x4(In[0],  In[1],  In[2],
+                      In[3],  In[4],  In[5],
+                      In[6],  In[7],  In[8],
+                      In[9],  In[10], In[11],
+                      In[12], In[13], In[14],
+                      In[15]);
+    
+    for (int i = 0; i < 4; i++) {
+        int4 B;
+        if (i % 2 == 0)
+            B = A[i];
```

Comment:
Just a comment, not a request for change:

These subscript accesses look dynamic, but these loops are likely to be unrolled, either before IR (in DXC, Clang, etc.) or after (in a driver compiler). That unrolling would make all accesses static, even eliminating the intermediate local variables to copy values directly from static indices of In to static indices of Out. That's a fine path to test, but this might not test other important dynamic indexing code paths, if desired.

I guess this gets at the fuzzy purpose issue with the offload-test-suite. This seems fine to test this scenario end-to-end, whatever path it may take through optimizations, and maybe that's all that's desired for this test.